distributed/tests/test_core.py

```diff
@@ -150,29 +152,58 @@ def listen_on(cls, *args):
             yield assert_can_connect(server.address)
             yield assert_can_connect_locally_6(server.port)
 
+    # InProc
 
-def test_rpc(loop):
-    @gen.coroutine
-    def f():
-        server = Server({'ping': pingpong})
-        server.listen(8883)
+    with listen_on(Server, 'inproc://') as server:
```

Comment:
The inproc stuff still seems pretty experimental to me.  I hesitate to integrate it into our core tests.  It would be nice to make sure that we can extract this cheaply in the future if we decide we don't want to keep it around.